### ISSUE TITLE
`map_unwrap_or` should suggest `is_ok_and(<f>)` on a result

### ISSUE THREAD

**Issue by xtqqczze** [rust-lang/rust-clippy#16258]
### Summary

https://godbolt.org/z/GbzzGqb9G

### Lint Name

`map_unwrap_or`

### Reproducer

I tried this code:

```rust
pub fn m1(result: Result<i32, ()>) -> bool {
    result
    .map(|x| x == 42)
    .unwrap_or(false)
}
```

I expected to see this happen:

```log
warning: called `map(<f>).unwrap_or(false)` on an `Result` value
...
help: use `is_ok_and(<f>)` instead
```

Instead, this happened:

no lint

### Version

```text
nightly
```

**Comment by profetia** [comment]
Will be covered in #15718
